src/mesh/coordinates.cxx

```diff
@@ -643,6 +643,10 @@ const Field2D Coordinates::Delp2(const Field2D &f) {
 const Field3D Coordinates::Delp2(const Field3D &f) {
   TRACE("Coordinates::Delp2( Field3D )");
 
+  if (localmesh->GlobalNx == 1 && localmesh->GlobalNz == 1) {
+    // copy mesh, location, etc
+    return f*0;
```

Comment:
Can this just be `return 0.0` or `return Field3D(0.0)` to avoid multiplication?

Reply:
No, it would need to be at least:
```
Field3D result(f.getMesh());
result=0;
result.setLocation(f.getLocation());
return result;
```
Maybe it is time to provide a simpler way to achieve copying all the data from the Field, except the data.

This would maybe work as well, as copying the field does not duplicate the data, only when the data is changed a new data block is allocated.
```
Field3D result = f;
result=0;
return result;
```
Anyway - I think the above multiplication shouldn't at all be an issue, as the field only has LocalNy points. In that case raw numerics is often cheap compared to all the other overheads ...